### PR TITLE
#2203: Channel Scrolling updates.

### PR DESCRIPTION
**Pull request by psymoon**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
- [x] This PR includes a **CHANGELOG entry** or does not need one
- [x] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

File: app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt [hunk 1]
```diff
@@ -26,6 +30,27 @@ fun View.validateKnownViewById(): Int {
...
+/**
+ * Set the horizontal margins on the given view.
+ *
+ * We want to add padding to the beginning and end of the RecyclerView: ideally we'd just add
+ * paddingStart/End. Unfortunately, this causes a visual glitch as each card scrolls offscreen.
+ * Instead, we set the margins for the first and last card.
+ *
+ * See [DefaultChannelAdapter.onBindViewHolder] for an example
+ */
+fun View.setHorizontalMargins(context: Context, position: Int, itemCount: Int) {
```
Comment (Baron-Severin) [inline app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: Does this need to be declared in both System and Gecko?  The implementations seem to be the same, unless I'm missing something.  Would it be possible to merge these?

Comment (Baron-Severin) [inline app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: This functionality doesn't seem broadly applicable to the rest of the app.  I also feel that this is too narrow a use case for it to make sense polluting the View namespace.  Could we move this into a companion object for `ChannelTile`, or something similar?

Comment (Baron-Severin) [inline app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: At first glance, I would expect a fun called `setHorizontalMargins` to take in two ints and set them to the left and right margins for a view.  Could we rename this to better communicate that it's handling a much more specific job?  `setChannelMarginByPosition` or something?  That's super wordy, but you see the direction I'm trying to move.

Comment (psymoon) [reply app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: View ext already exists in both variants, moving to main would raise a compiler issue

Comment (mcomella) [reply app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: I don't understand: what compiler issue would you see?

Comment (mcomella) [reply app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: Nvm, looks like this was removed anyway.

Comment (Baron-Severin) [reply app/src/gecko/java/org/mozilla/tv/firefox/ext/View.kt]: @mcomella At any given time, there would have been two files with the same package statement & name (one from main, one from either system or GV).

**Pull request merged**